sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/csv/UnivocityParser.scala

```diff
@@ -346,4 +348,23 @@ private[sql] object UnivocityParser {
       parser.options.columnNameOfCorruptRecord)
     filteredLines.flatMap(safeParser.parse)
   }
+
+  def limitParserErrorContentLength[T](f: () => T): T = {
+    try {
+      f()
+    } catch {
+      case e: TextParsingException =>
+        e.setErrorContentLength(SQLConf.get.getConf(
```

Comment:
@WeichenXu123, seems `setErrorContentLength` can be set in `CSVOptions`'s parser and writer settings (see https://github.com/uniVocity/univocity-parsers/blob/f616d151b48150bc9cb98943f9b6f8353b704359/src/test/java/com/univocity/parsers/common/DataProcessingExceptionTest.java)

Reply:
Oh you're right. I didn't realize this.
So that we can direct set the parser/writer option.
Btw, should this be added as a csv datasource option ?

Reply:
I think we can just add an option without a configuration for now.

Reply:
We should fix readwriter.py, DataStreamReader/Writer DataFrameReader/Writer too when we add an option fyi.